lockup-factory/src/lib.rs

```diff
@@ -0,0 +1,423 @@
+    foundation_account_id: Option<AccountId>,
+}
+
+
+impl Default for LockupFactory {
+    fn default() -> Self {
+        env::panic(b"LockupFactory should be initialized before usage")
+    }
+}
+
+#[near_bindgen]
+impl LockupFactory {
+    #[init]
+    pub fn new(master_account_id: AccountId,
+               lockup_master_account_id: AccountId,
+               whitelist_account_id: AccountId,
+               foundation_account_id: AccountId) ->
+               Self {
+        assert!(!env::state_exists(), "The contract is already initialized");
+
+        assert!(
+            env::is_valid_account_id(master_account_id.as_bytes()),
+            "The master account ID is invalid"
+        );
+
+        assert!(
+            env::is_valid_account_id(lockup_master_account_id.as_bytes()),
+            "The lockup account ID is invalid"
+        );
+        assert!(
+            env::is_valid_account_id(whitelist_account_id.as_bytes()),
+            "The whitelist account ID is invalid"
+        );
+
+        assert!(
+            env::is_valid_account_id(foundation_account_id.as_bytes()),
+            "The foundation account is invalid"
+        );
+        Self {
+            master_account_id,
+            lockup_master_account_id,
+            whitelist_account_id,
+            foundation_account_id,
+        }
+    }
+
+    /// Returns the foundation account id.
+    pub fn get_foundation_account_id(&self) -> String {
+        self.foundation_account_id.to_string()
+    }
+
+    /// Returns the master account id.
+    pub fn get_master_account_id(&self) -> String {
+        self.master_account_id.to_string()
+    }
+
+    /// Returns the lockup account id.
+    pub fn get_lockup_master_account_id(&self) -> String {
+        self.lockup_master_account_id.to_string()
+    }
+
+    /// Returns minimum attached balance.
+    pub fn get_min_attached_balance(&self) -> U128 {
+        MIN_ATTACHED_BALANCE.into()
+    }
+
+
+    #[payable]
+    pub fn create(&mut self,
+                  owner_account_id: AccountId,
+                  lockup_duration: WrappedDuration,
+                  lockup_timestamp: Option<WrappedTimestamp>,
+                  vesting_schedule: Option<VestingScheduleOrHash>,
+                  release_duration: Option<WrappedDuration>,
+    ) -> Promise {
+        assert!(
+            env::attached_deposit() >= MIN_ATTACHED_BALANCE,
+            "Not enough attached deposit"
+        );
+
+        assert!(
+            env::is_valid_account_id(owner_account_id.as_bytes()),
+            "The owner account ID is invalid"
+        );
+
+        let byte_slice = env::sha256(owner_account_id.as_bytes());
+        let string = byte_slice.iter().map(|x| format!("{:02x}", x)).collect::<String>();
+        let lockup_suffix = ".".to_string() + &self.lockup_master_account_id.to_string();
+        let sliced_string = &string[..40];
+        let lockup_account_id: AccountId = sliced_string.to_owned() + &lockup_suffix;
```

Comment:
Using `hex` and `format` 
```suggestion
        let lockup_account_id = format!(
            "{}.{}",
            hex::encode(&byte_slice[..20]),
            self.lockup_master_account_id,
        )
```